zulip_bots/zulip_bots/simple_lib.py

```diff
@@ -39,15 +39,22 @@ def update(self, message):
     def upload_file(self, file):
         return dict(result='success', msg='', uri='https://server/user_uploads/{}'.format(uuid4()))
 
-class TerminalBotHandler:
+class TerminalBotHandler(BotHandler):
     def __init__(self, bot_config_file):
```

Comment:
I don't think `TerminalBotHandler` needs to inherit `BotHandler`, by default all `BotHandler`s follow the protocol.

Reply:
Fixed that.

Reply:
Hmm, I kinda feel like this should inherit from AbstractBotHandler once we do that rename.

Reply:
If we inherit a protocol, all properties of the protocol now come into the class even if the class does not implement it. Since the implementation of abstract functions is not a mandate. I would rather recommend creating functions and not inheriting them so that we can come back to them later. Inheriting a Protocol also skips a lot of mypy issues like the ones we faced here which gets caught, if we keep them separate.

Reply:
I've not looked deeply at these recent changes but if this is intended to be a `Protocol` (~ static duck typing), then I agree with @LoopThrough-i-j re not inheriting. It may make it clearer to add a `Protocol` suffix to `BotHandler` to indicate that.